### PR TITLE
SAP-6360 Added XRegExp to handle unicode characters

### PR DESCRIPTION
**Pull request by Chris-Congenica**
This is basically an update to the existing regex to use better matching for unicode characters so that chinese don't get filtered out.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,8 +36,9 @@
   "dependencies": {
     "jquery": "^3.1.1",
     "lodash": "^4.17.4",
-    "prototype": "timrichd/prototype.node.js",
+    "prototype": "github:timrichd/prototype.node.js",
```
Comment (congenicamark) [inline package.json]: Does this mean that we're reliant on an ex employee not removing their github account?

File: src/hpoTerm.js [hunk 2]
```diff
@@ -69,7 +71,9 @@ HPOTerm.sanitizeID = function(id) {
   temp = temp.replace(/[\)\]]/g, '_J_');
   temp = temp.replace(/[:]/g, '_C_');
   temp = temp.replace(/\,/g, '__COMMA__');
-  return temp.replace(/[^a-zA-Z0-9,;_\-*]/g, '__');
+  // XRegExp handles unicode characters for us, so languages like chinese don't
+  // get filtered out here.
+  return temp.replace(XRegExp('[^,;_\\p{Dash_Punctuation}*\\pL]', 'g'),  '__');
```
Comment (congenicamark) [inline src/hpoTerm.js]: I'd rather see the \\pL (biggest, most important set), then \\p{Dash_Punctuation} (next grouping), then all of the punctuation chars grouped next to each other. At the moment, it's a bit disorganised. Functionally looks fine though, so am approving.

**Review by congenicamark — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Chris-Congenica) [reply package.json]: Looks like it ye, I'll see if I can get a separate repo set up and fork Tim's repo.

Comment (Chris-Congenica) [reply package.json]: Actually, Tim's fork is the same as the master of the original. I'll update this to just use the npm package.

Comment (Chris-Congenica) [reply package.json]: But if only the author had actually updated npm with the latest version of that package... https://github.com/Rixius/prototype.node.js/issues/5

File: src/hpoTerm.js (hunk 2 above)
Comment (Chris-Congenica) [reply src/hpoTerm.js]: Fair point, I've re-organised things.

File: package.json (hunk 1 above)
Comment (Chris-Congenica) [reply package.json]: We now have a public fork at: https://github.com/Congenica/prototype.node.js

**Pull request merged**